ui/src/app/workflows/components/workflow-filters/workflow-filters.tsx

```diff
@@ -57,13 +90,45 @@ export function WorkflowFilters(props: WorkflowFilterProps) {
         return results;
     }, [props.workflows, props.phaseItems]);
 
+    const handleNameFilterChange = (item: {title: string; id: string}) => {
+        props.setNamePrefix('');
+        props.setNamePattern('');
+        props.setName('');
+        setNameFilter(item);
+    };
+
     return (
         <div className='wf-filters-container'>
             <div className='row'>
                 <div className='columns small-2 xlarge-12'>
                     <p className='wf-filters-container__title'>Namespace</p>
                     <NamespaceFilter value={props.namespace} onChange={props.setNamespace} />
                 </div>
+                <div className='columns small-2 xlarge-12'>
+                    <DropDown
+                        isMenu
+                        closeOnInsideClick
+                        anchor={
+                            <div className={classNames('top-bar__filter')} title='Filter'>
+                                <p className='wf-filters-container__title'>
+                                    {nameFilter.title} <i className='fa fa-angle-down' aria-hidden='true' />
+                                </p>
+                            </div>
+                        }>
+                        <ul id='top-bar__filter-list'>
+                            {NAME_FILTERS.map((item, i) => (
+                                <li key={i} className={classNames('top-bar__filter-item', {title: true})} onClick={() => handleNameFilterChange(item)}>
+                                    <span>{item.title}</span>
+                                </li>
+                            ))}
+                        </ul>
+                    </DropDown>
+                    {nameFilter.id === 'namePrefix' ? <InputFilter value={props.namePrefix} name='wfNamePrefix' onChange={props.setNamePrefix} placeholder='Search...' /> : null}
+                    {nameFilter.id === 'namePattern' ? (
+                        <InputFilter value={props.namePattern} name='wfNamePattern' onChange={props.setNamePattern} placeholder='Search...' />
+                    ) : null}
+                    {nameFilter.id === 'name' ? <InputFilter value={props.name} name='wfName' onChange={props.setName} placeholder='Search...' /> : null}
```

Comment:
this can be simplified and made neater by having an `if` statement above and changing the `value` and `onChange` based on that

also placeholders [should show example values](https://ux.stackexchange.com/a/131591/86443)

Reply:
actually the below comment will simplify this even more